test/provfetchtest.c

```diff
@@ -84,6 +86,95 @@ static const OSSL_ALGORITHM dummy_store[] = {
     { NULL, NULL, NULL }
 };
 
+static void *dummy_rand_newctx(void *provctx, void *parent,
```

Comment:
Why not reuse the _fake_random_ that already exists in the tests?

Expose `fake_rand_functions`, `fake_rand_rand` or `fake_rand_query` and call from the dummy provider.

Reply:
Since this is already approved, I'm going to leave this suggestion for a possible future improvement.